src/dbnode/storage/index.go

```diff
@@ -1580,8 +1580,11 @@ func (i *nsIndex) queryWithSpan(
 			break
 		}
 
+		// Calculate time spent waiting for a worker
+		waitSpan, _ := ctx.StartTraceSpan(tracepoint.NSIdxQueryWait)
```

Comment:
seems like this would be generally useful inside the workerpool

Reply:
yeah good idea. Updated (but this means `GoWithContext` takes an m3 context 😬